src/App.tsx

```
@@ -5,7 +5,19 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const onAdd = (movie: Movie) => {
+    const hasMovie = movies
+      .find(movieFromList => movieFromList.imdbId
+        === movie.imdbId);
+
+    if (hasMovie) {
+      return;
```

you can throw an error here